cnf-certification-test/accesscontrol/rbac/roles.go

```diff
@@ -65,14 +71,20 @@ func GetAllRules(aRole *rbacv1.Role) (ruleList []RoleRule) {
 	return ruleList
 }
 
-// Checks the resource name in the role against plural name
+// isResourceInRoleRule Checks the resource name in the role against plural name
+// Returns:
+//   - bool : if a CrdResource matches a RoleRule based on their properties return true , otherwise return false.
 func isResourceInRoleRule(crd CrdResource, roleRule RoleRule) bool {
 	// remove subresources to keep only resource (plural) name
 	ruleResourcePluralName := strings.Split(roleRule.Resource.Name, "/")[0]
 
 	return crd.Group == roleRule.Resource.Group && crd.PluralName == ruleResourcePluralName
 }
 
+// FilterRulesNonMatchingResources filters RoleRules based on whether they match any CrdResource in the resourceList.
+// Returns :
+//   - Matching: a slice of RoleRule that contains all rules where a CrdResource matches a RoleRule based on their properties.
+//   - NonMatching: a slice of RoleRule that contains all rules present in ruleList but not matching.
```

Comment:
- NonMatching: a slice of RoleRule that contains all rules present in ruleList but not matching.
->
- NonMatching: a slice of RoleRule that contains all rules not matching the CRD resource